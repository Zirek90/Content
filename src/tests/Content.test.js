import React from "react";
import renderer from "react-test-renderer";
import ContentComponent from "../components/Content";

describe("ContentComponent", () => {
  it("renders correctly", () => {
    const props = {
      author: {
        value: "John Doe",
      },
      date: {
        value: "2020-09-06T22:00:00Z",
      },
      body: {
        values: [
          "<h2><span>Hey there</span></h2> ",
          "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p> ",
        ],
      },
    };
    const tree = renderer.create(<ContentComponent content={props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
