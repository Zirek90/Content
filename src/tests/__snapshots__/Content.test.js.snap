// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentComponent renders correctly 1`] = `
<div
  className="sc-jSFkmK WjmPB"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2><span>Hey there</span></h2> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p> ",
      }
    }
  />
  <div>
    <p
      className="sc-gtssRu iBDuNi"
    >
      John Doe
    </p>
    <p
      className="sc-gtssRu kZfLwQ"
    >
      Sun, 06 Sep 2020 22:00:00 GMT
    </p>
  </div>
</div>
`;
